executor/distsql.go

```diff
@@ -361,25 +366,78 @@ func (e *TableReaderExecutor) Open(goCtx goctx.Context) error {
 	span, goCtx := startSpanFollowsContext(goCtx, "executor.TableReader.Open")
 	defer span.Finish()
 
+	e.resultHandler = &tableResultHandler{}
+	firstPartRanges, secondPartRanges := e.splitRanges()
+	firstResult, err := e.buildResp(goCtx, firstPartRanges)
+	if err != nil {
+		e.feedback.Invalidate()
+		return errors.Trace(err)
+	}
+	if len(secondPartRanges) == 0 {
+		e.resultHandler.open(nil, firstResult)
```

Comment:
Why not always put `firstResult` at the first position of arguments?

Reply:
This way we can alway guarantee the second result exists so don't need to check whether second result exists.

Reply:
Why not always guarantee the first result exists?

Reply:
If the optional one is first. We only need to check whether the first one is ended.
Otherwise we need to check whether the first one is ended and whether the second one exists.